app/api/validators.py

```
@@ -0,0 +1,351 @@
+                # Basic check to prevent injection but allow emails
+                if '<' in value or '>' in value or ';' in value or '\'' in value or '"' in value:
+                    raise HTTPException(422, f"value for '{safe_key}' contains invalid characters")
+                sanitized[safe_key] = value.strip()
+            else:
+                sanitized[safe_key] = validate_safe_string(value, f"value for '{safe_key}'", allow_empty=True)
+        elif isinstance(value, dict):
+            sanitized[safe_key] = sanitize_dict(value, max_depth - 1)
+        elif isinstance(value, list):
+            sanitized[safe_key] = [
+                sanitize_dict(item, max_depth - 1) if isinstance(item, dict)
+                else validate_safe_string(item, f"list item", allow_empty=True) if isinstance(item, str) and '@' not in item
```

Ruff (0.12.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)